path_login.go

```diff
@@ -170,30 +175,58 @@ func (b *azureAuthBackend) verifyClaims(claims *additionalClaims, role *azureRol
 	return nil
 }
 
-func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, resourceGroupName, vmName string, claims *additionalClaims, role *azureRole) error {
+func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, resourceGroupName, vmName string, vmssName string, claims *additionalClaims, role *azureRole) error {
 	// If not checking anything with the resource id, exit early
 	if len(role.BoundResourceGroups) == 0 && len(role.BoundSubscriptionsIDs) == 0 && len(role.BoundLocations) == 0 {
 		return nil
 	}
 
-	if subscriptionID == "" || resourceGroupName == "" || vmName == "" {
-		return errors.New("subscription_id, resource_group_name, and vm_name are required")
+	if subscriptionID == "" || resourceGroupName == "" || (vmName == "" && vmssName == "") {
+		return errors.New("subscription_id, resource_group_name, and either vm_name or vmss_name are required")
 	}
 
-	client := b.provider.ComputeClient(subscriptionID)
-	vm, err := client.Get(ctx, resourceGroupName, vmName, compute.InstanceView)
-	if err != nil {
-		return errwrap.Wrapf("unable to retrieve virtual machine metadata: {{err}}", err)
+	var principalID, location *string
+
+	if vmssName != "" {
+		client := b.provider.VMSSClient(subscriptionID)
+		vmss, err := client.Get(ctx, resourceGroupName, vmssName)
+
+		if err != nil {
+			return errwrap.Wrapf("unable to retries virtual machine scale set metadata: {{err}}", err)
+		}
+
+		if vmss.Identity == nil {
```

Comment:
I haven't checked but do vms and vmss share the same identity type?  If so, you can just pull out the identity and share the logic between these two methods.

Reply:
Sadly they don't. One is a VirtualMachineIdentity and the other is VirtualMachineScaleSetIdentity - although (in the current SDK) they do have identical contents.

Reply:
No problem. This works.